gatsby-browser.tsx

```diff
@@ -40,10 +45,40 @@ const shouldUpdateScroll: GatsbyBrowser['shouldUpdateScroll'] = ({ prevRouterPro
   );
 };
 
-/**
- * Load our user state
- */
-import UserContextWrapper from './src/contexts/user-context/wrap-with-provider';
-const wrapRootElement = UserContextWrapper;
+const InsightsWrapper = ({ children }) => {
+  // Use a ref to track if we've already initialized
+  const initialized = useRef(false);
+  const { externalScriptsData } = useSiteMetadata();
+
+  useEffect(() => {
+    // Initialize insights only once
+    if (!initialized.current) {
+      initialized.current = true;
+
+      initInsights({
+        debug: externalScriptsData.insightsDebug,
+        mixpanelToken: externalScriptsData.mixpanelApiKey,
+        mixpanelAutoCapture: externalScriptsData.mixpanelAutoCapture,
+        posthogApiKey: externalScriptsData.posthogApiKey,
+        posthogApiHost: externalScriptsData.posthogHost,
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Was this `externalScriptsData`?

Reply:
@jamiehenson same as we had in Voltaire:

`
React Hook useEffect has missing dependencies: 'externalScriptsData.insightsDebug', 'externalScriptsData.mixpanelApiKey', 'externalScriptsData.mixpanelAutoCapture', 'externalScriptsData.posthogApiKey', and 'externalScriptsData.posthogHost'. Either include them or remove the dependency array.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)
`

Reply:
(and these things don't change, they are set at build time)